compiler/rustc_passes/src/liveness.rs

```diff
@@ -1173,7 +1173,7 @@ impl<'a, 'tcx> Liveness<'a, 'tcx> {
                             }
                         }
                         hir::InlineAsmOperand::InOut { expr, .. } => {
-                            succ = self.write_place(expr, succ, ACC_READ | ACC_WRITE);
+                            succ = self.write_place(expr, succ, ACC_READ | ACC_WRITE | ACC_USE);
```

Comment:
Just a note: changing line 1197 to `succ = self.propagate_through_expr(expr, succ);` would have the same effect as this change. I'm not entirely sure which change is more semantically correct in the context of the pass- willing to change this if anyone knows better than me.

Reply:
I think that I would prefer updating the match below.

Reply:
Thanks- changed in the latest commit.

Reply:
Both write_place(expr) and propagate_through_expr(expr) will use the same live node for path expressions. Using them with the same expression is incorrect, since they will overwrite the previous content. In this case propagate_through_expr will access with READ & USED, overwriting WRITE set by write_place, which will lead to a false negative if a result of inout is in fact unused and should be warned about.

The inout case is quite unlike anything else here, but the initial version looked correct to me. A few additional positive & negative test cases should clarify the situation.

Reply:
`acc` looks like it's specifically not overwriting in the WRITE then READ & USED case.

Reply:
Hmm. The issue is just limited to the incorrect successor then.

Reply:
I see. The original change is the only correct fix here then.

Reply:
Reverted to the original change- thanks for digging into this, y'all